app/domain/operations/hbx_enrollments/drop_enrollment_members.rb

```diff
@@ -11,51 +11,101 @@ class DropEnrollmentMembers
       def call(params)
         values          = yield validate(params)
         dropped_members = yield drop_enrollment_members(values)
+
         Success(dropped_members)
       end
 
       private
 
       def validate(params)
+        return Failure('Member drop feature is turned off.') unless EnrollRegistry.feature_enabled?(:drop_enrollment_members)
         return Failure('Missing HbxEnrollment Key.') unless params.key?(:hbx_enrollment)
         return Failure('Not a valid HbxEnrollment object.') unless params[:hbx_enrollment].is_a?(HbxEnrollment)
+        return Failure('Enrollment need be in an active state to drop dependent') unless params[:hbx_enrollment].is_admin_terminate_eligible?
+        return Failure('Not an ivl enrollment.') unless params[:hbx_enrollment].is_ivl_by_kind? # Added per the request, also need further development if to be used for shop
         return Failure('No members selected to drop.') if params[:options].select{|string| string.include?("terminate_member")}.empty?
         return Failure('No termination date given.') if params[:options].select{|string| string.include?("termination_date")}.empty?
+        return Failure('Termination date cannot be outside of the current calender year.') unless Date.strptime(params[:options]["termination_date_#{params[:hbx_enrollment].id}"], "%m/%d/%Y").year == TimeKeeper.date_of_record.year
 
         Success(params)
       end
 
       def drop_enrollment_members(params)
-        terminate_date = Date.strptime(params[:options]["termination_date_#{params[:hbx_enrollment].id}"], "%m/%d/%Y")
+        termination_date = Date.strptime(params[:options]["termination_date_#{params[:hbx_enrollment].id}"], "%m/%d/%Y")
         dropped_enr_members = params[:options].select{|string| string.include?("terminate_member")}.values
         all_enr_members = params[:hbx_enrollment].hbx_enrollment_members
-        eligible_members = all_enr_members.reject{ |member| dropped_enr_members.include?(member.id.to_s) }
-
-        reinstatement = Enrollments::Replicator::Reinstatement.new(params[:hbx_enrollment], terminate_date + 1.day, 0, eligible_members).build
-        reinstatement.save!
+        non_eligible_members = all_enr_members.select{ |member| dropped_enr_members.include?(member.id.to_s) }
+        new_effective_date = (termination_date > params[:hbx_enrollment].effective_on) ? termination_date + 1.day : params[:hbx_enrollment].effective_on
+        return Failure('Select termination date that would result member drop in present calender year.') unless new_effective_date.year == termination_date.year
```

Comment:
can you move this logic to validate method so that all validations will be in one place.